notify/rocketchat/rocketchat.go

```diff
@@ -0,0 +1,266 @@
+func New(c *config.RocketchatConfig, t *template.Template, l log.Logger, httpOpts ...commoncfg.HTTPClientOption) (*Notifier, error) {
+	client, err := commoncfg.NewClientFromConfig(*c.HTTPConfig, "rocketchat", httpOpts...)
+	if err != nil {
+		return nil, err
+	}
+	token, err := getToken(c)
+	if err != nil {
+		return nil, err
+	}
+	tokenID, err := getTokenID(c)
+	if err != nil {
+		return nil, err
+	}
+	return &Notifier{
+		conf:         c,
+		tmpl:         t,
+		logger:       l,
+		client:       client,
+		retrier:      &notify.Retrier{},
+		postJSONFunc: notify.PostJSON,
+		token:        token,
+		tokenID:      tokenID,
+	}, nil
+}
+
+func getTokenID(c *config.RocketchatConfig) (string, error) {
+	if len(c.TokenIDFile) > 0 {
+		content, err := os.ReadFile(c.TokenIDFile)
+		if err != nil {
+			return "", fmt.Errorf("could not read %s: %w", c.TokenIDFile, err)
+		}
+		return strings.TrimSpace(string(content)), nil
+	}
+	return string(*c.TokenID), nil
+}
+
+func getToken(c *config.RocketchatConfig) (string, error) {
+	if len(c.TokenFile) > 0 {
+		content, err := os.ReadFile(c.TokenFile)
+		if err != nil {
+			return "", fmt.Errorf("could not read %s: %w", c.TokenFile, err)
+		}
+		return strings.TrimSpace(string(content)), nil
+	}
+	return string(*c.Token), nil
+}
+
+// Notify implements the Notifier interface.
+func (n *Notifier) Notify(ctx context.Context, as ...*types.Alert) (bool, error) {
+	var err error
+
+	data := notify.GetTemplateData(ctx, n.tmpl, as, n.logger)
+	tmplText := notify.TmplText(n.tmpl, data, &err)
+	if err != nil {
+		return false, err
+	}
+	titlex := tmplText(n.conf.Title)
+	if err != nil {
+		return false, err
+	}
+
+	title, truncated := notify.TruncateInRunes(titlex, maxTitleLenRunes)
+	if truncated {
+		key, err := notify.ExtractGroupKey(ctx)
+		if err != nil {
+			return false, err
+		}
+		level.Warn(n.logger).Log("msg", "Truncated title", "key", key, "max_runes", maxTitleLenRunes)
+	}
+	att := &Attachment{
+		Title:     title,
+		TitleLink: tmplText(n.conf.TitleLink),
+		Text:      tmplText(n.conf.Text),
+		ImageURL:  tmplText(n.conf.ImageURL),
+		ThumbURL:  tmplText(n.conf.ThumbURL),
+		Color:     tmplText(n.conf.Color),
+	}
+	numFields := len(n.conf.Fields)
+	if numFields > 0 {
+		fields := make([]config.RocketchaAttachmentField, numFields)
+		for index, field := range n.conf.Fields {
+			// Check if short was defined for the field otherwise fallback to the global setting
+			var short bool
+			if field.Short != nil {
+				short = *field.Short
+			} else {
+				short = n.conf.ShortFields
+			}
+
+			// Rebuild the field by executing any templates and setting the new value for short
+			fields[index] = config.RocketchaAttachmentField{
+				Title: tmplText(field.Title),
+				Value: tmplText(field.Value),
+				Short: &short,
+			}
+		}
+		att.Fields = fields
+	}
+	numActions := len(n.conf.Actions)
+	if numActions > 0 {
+		actions := make([]config.RocketchatAttachmentAction, numActions)
```

Comment:
Same here, do we need to enforce limits?

Reply:
see above